### ISSUE TITLE
FetchContent doesn't seem to work as described in the documentation with Catch2

### ISSUE THREAD

**Issue by tjwrona** [catchorg/Catch2#2103]
I am trying to use `FetchContent` to include Catch2 into a CMake project I am working on, but it does not seem to work. After using `FetchContent` as described in the documentation the `catch_discover_tests()` function is still not found.

Following the documentation it states that `FetchContent` should be usable for including Catch2 into a project as follows:

```cmake
Include(FetchContent)

FetchContent_Declare(
  Catch2
  GIT_REPOSITORY https://github.com/catchorg/Catch2.git
  GIT_TAG        v2.13.1)

FetchContent_MakeAvailable(Catch2)

target_link_libraries(tests Catch2::Catch2)
```

But after following these steps, I am still unable to call `catch_discover_tests()` within my CMake files.

I even tried adding `find_package(Catch2 REQUIRED)` and `include(Catch)` but both of those commands fail as well.

**Platform information:**

 - OS: **Windows 10**
 - Compiler+version: **Visual Studio 2019 16.19.0 Preview 1.0**
 - Catch version: **v2.13.3**

**Comment by David-Durst** [comment]
I think the following should work:
```
Include(FetchContent)

FetchContent_Declare(
  Catch2
  GIT_REPOSITORY https://github.com/catchorg/Catch2.git
  GIT_TAG        v2.13.1)

FetchContent_MakeAvailable(Catch2)

target_link_libraries(tests Catch2::Catch2)

list(APPEND CMAKE_MODULE_PATH ${catch2_SOURCE_DIR}/contrib)
include(CTest)
include(Catch)
catch_discover_tests()
```

Per https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md#automatic-test-registration, if Catch2 has been downloaded but not installed (i.e. what FetchContent_Declare and FetchContent_MakeAvailable does), then "you need to add them to your CMake module path."

It seems that since FetchContent makes the package available during configuration time and before building, the *.cmake files aren't created yet and are still in *.cmake.in form in https://github.com/catchorg/Catch2/tree/devel/CMake.

If a project maintainer has a better solution, let me know!

**Comment by shaperilio** [comment]
> ```
> Include(FetchContent)
> 
> FetchContent_Declare(
>   Catch2
>   GIT_REPOSITORY https://github.com/catchorg/Catch2.git
>   GIT_TAG        v2.13.1)
> 
> FetchContent_MakeAvailable(Catch2)
> 
> target_link_libraries(tests Catch2::Catch2)
> 
> list(APPEND CMAKE_MODULE_PATH ${catch2_SOURCE_DIR}/contrib)
> include(CTest)
> include(Catch)
> catch_discover_tests()
> ```
> 
> Per https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md#automatic-test-registration, if Catch2 has been downloaded but not installed (i.e. what FetchContent_Declare and FetchContent_MakeAvailable does), then "you need to add them to your CMake module path."

Adding it to the path is a critical step that should be included in the sample CMakeLists in the docs.

There are many things obvious to experienced folks but not so for people trying to get started.

For example:
[This](https://github.com/catchorg/Catch2/blob/v2.x/docs/tutorial.md#getting-catch2) implies you can either download the single header *or* install via package manager / build from source / download via CMake. The CMake solution is missing appending it to the path as detailed in @David-Durst's reply. The install via package manager / build from source is missing the fact that you must then add the include directory to be able to find `<catch2/catch.hpp>`. A solution is to both download the single header *and* e.g. build from source, at which point your test files can have `#include "path/to/catch.hpp" and your `CMakeLists.txt` can have

```
find_package(Catch2 REQUIRED)
target_link_libraries(test Catch2::Catch2)
```

The documentation [here](https://github.com/catchorg/Catch2/blob/v2.x/docs/cmake-integration.md#cmake-target) seems to imply this is enough to be able to write `#include <catch2/catch.hpp>` and not download the header separately, but that's obviously not the case; you cannot compile anything if it can't find the include files, so how could `target_link_libraries` fix that?

Having examined `install_manifest.txt` after installing from source following [this](https://github.com/catchorg/Catch2/blob/v2.x/docs/cmake-integration.md#installing-catch2-from-git-repository) I see that the actual problem is that there is no file called `catch.hpp`. If I instead use `catch_all.hpp`, the linker fails as no `main` is defined.

So I have to write my own "do-nothing" main by following the example [here](https://github.com/catchorg/Catch2/blob/devel/docs/own-main.md#let-catch-take-full-control-of-args-and-config)

In summary, I believe there are both issues with the documentation (e.g. `#include <catch2/catch.hpp>` will never work because that file doesn't exist) and a lot of confusion for inexperienced folks like me from trying to assemble things from fractional pieces.

It seems that between what I've found and what @David-Durst has pointed out, some changes should be made.

**Comment by zekrom-vale** [comment]
Thank you so much!  This fixed all the issues I was having with CMake and Catch2.

**Comment by stolpa4** [comment]
Struggling the whole day with this. Is it possible to use the FetchContent with catch2 at all? I've tried everything I found on the internet - nothing works. CMake just can't find the catch.hpp header
